### PR TITLE
Pjdotson/sy-746 Console: Add renaming for cluster and range toolbars

### PR DESCRIPTION
**Pull request by pjdotson**
Added renaming ability for ranges and clusters through their respective toolbars.

### CODE REVIEW

**Comment by pjdotson** [comment]
# Feature Pull Request Template

## Key Information

- **Linear Issue**: [SY-746](https://linear.app/synnaxlabs/issue/SY-746/add-renaming-to-cluster-and-range-toolbars)

## Description

Added renaming for cluster and ranges in the toolbar. Also some minor refactoring done to other parts of the code.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes.
- [x] I have updated user facing documentation accordingly.

## Migrations

- [x] Console - I have ensured that previous versions of stored data structures are 
properly migrated to new formats.
- [x] Server - I have ensured that previous versions of stored data structures are 
properly migrated to new formats.

## Manual QA Additions

- [x] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
with necessary manual QA steps to test my change.

## Breaking Changes

Please list any breaking changes to public or internal packages.

File: .github/PULL_REQUEST_TEMPLATE/rc.md
Comment (emilbon99) [inline .github/PULL_REQUEST_TEMPLATE/rc.md:1]: What about renaming a workspace and a cluster?

File: console/src/workspace/ontology.tsx [hunk 1]
```diff
@@ -115,9 +114,8 @@ const TreeContextMenu: Ontology.TreeContextMenu = (props): ReactElement => {
     void Group.fromSelection(props);
   };
 
-  const clusterKey = Cluster.useSelectActiveKey();
   const handleLink = (): void => {
-    const toCopy = `synnax://cluster/${clusterKey}/workspace/${selection.resources[0].id.key}`;
+    const toCopy = `synnax://cluster/${client.key}/workspace/${selection.resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/workspace/ontology.tsx:118]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/schematic/ontology.tsx [hunk 2]
```diff
@@ -98,9 +97,8 @@ const TreeContextMenu: Ontology.TreeContextMenu = ({
 
   const handleRename = (): void => Tree.startRenaming(resources[0].key);
 
-  const clusterKey = Cluster.useSelectActiveKey();
   const handleCopyUrl = (): void => {
-    const url = `synnax://cluster/${clusterKey}/schematic/${resources[0].id.key}`;
+    const url = `synnax://cluster/${client.key}/schematic/${resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/schematic/ontology.tsx:101]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/range/slice.ts [hunk 3]
```diff
@@ -94,22 +98,27 @@ export const { actions, reducer } = createSlice({
...
+    clearBuffer: (state) => {
+      state.buffer = null;
+    },
     remove: (state, { payload: { keys } }: PA<RemovePayload>) => {
-       
       keys.forEach((k) => delete state.ranges[k]);
     },
+    rename: (state, { payload: { key, name } }: PA<RenamePayload>) => {
+      const range = state.ranges[key];
+      if (range != null) {
```
Comment (emilbon99) [inline console/src/range/slice.ts:109]: Use an early return here or remove the braces

File: console/src/range/accordionEntry.tsx [hunk 4]
```diff
@@ -104,6 +104,10 @@ export const List = (): ReactElement => {
     );
   };
 
+  const handleRename = (key: string): void => {
```
Comment (emilbon99) [inline console/src/range/accordionEntry.tsx:107]: No need for braces here

File: console/src/lineplot/ontology.tsx [hunk 5]
```diff
@@ -38,12 +37,10 @@ const TreeContextMenu: Ontology.TreeContextMenu = ({
     })();
   };
 
-  const clusterKey = Cluster.useSelectActiveKey();
-
   const handleRename = (): void => Tree.startRenaming(resources[0].key);
 
   const handleCopyLink = (): void => {
-    const toCopy = `synnax://cluster/${clusterKey}/lineplot/${resources[0].id.key}`;
+    const toCopy = `synnax://cluster/${client.key}/lineplot/${resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/lineplot/ontology.tsx:43]: Use `clusterKey` not `client.key`

File: console/src/hardware/task/ontology.tsx [hunk 6]
```diff
@@ -74,7 +72,7 @@ const TreeContextMenu: Ontology.TreeContextMenu = (props) => {
         });
         break;
       case "link": {
-        const url = `synnax://cluster/${clusterKey}/task/${selection.resources[0].id.key}`;
+        const url = `synnax://cluster/${client.key}/task/${selection.resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/hardware/task/ontology.tsx:75]: Use `clusterKey` and not `client.key`

File: console/src/range/ontology.tsx [hunk 7]
```diff
@@ -181,7 +181,7 @@ const TreeContextMenu: Ontology.TreeContextMenu = (props) => {
         void Group.fromSelection(props);
         return;
       case "link": {
-        const toCopy = `synnax://cluster/${clusterKey}/range/${resources[0].id.key}`;
+        const toCopy = `synnax://cluster/${client.key}/range/${resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/range/ontology.tsx:184]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/lineplot/ontology.tsx (hunk 5 above)
Comment (emilbon99) [reply console/src/lineplot/ontology.tsx:43]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/hardware/task/ontology.tsx (hunk 6 above)
Comment (emilbon99) [reply console/src/hardware/task/ontology.tsx:75]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/hardware/device/ontology.tsx [hunk 8]
```diff
@@ -79,7 +77,7 @@ const TreeContextMenu: Ontology.TreeContextMenu = (props) => {
         handleDelete(props);
         break;
       case "link": {
-        const toCopy = `synnax://cluster/${clusterKey}/device/${selection.resources[0].id.key}`;
+        const toCopy = `synnax://cluster/${client.key}/device/${selection.resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/hardware/device/ontology.tsx:80]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

File: console/src/cluster/Dropdown.tsx [hunk 9]
```diff
@@ -163,12 +185,19 @@ const ListItem = (props: CoreList.ItemProps<string, Cluster>): ReactElement => {
         break;
     }
   }
+
+  const handleChange = (value: string) => {
+    if (value.length === 0) return;
```
Comment (emilbon99) [inline console/src/cluster/Dropdown.tsx:190]: It shouldn't be possible for this value to be of length 0, `Text.Editable` prevents it. Can we remove?

File: console/src/channel/ontology.tsx [hunk 10]
```diff
@@ -185,7 +183,7 @@ const TreeContextMenu: Ontology.TreeContextMenu = (props) => {
         void Group.fromSelection(props);
         break;
       case "link": {
-        const toCopy = `synnax://cluster/${clusterKey}/channel/${resources[0].id.key}`;
+        const toCopy = `synnax://cluster/${client.key}/channel/${resources[0].id.key}`;
```
Comment (emilbon99) [inline console/src/channel/ontology.tsx:186]: This won't work. The client key is not equal to the cluster key. Use the cluster key instead.

**Review by emilbon99 — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE/rc.md
Comment (pjdotson) [reply .github/PULL_REQUEST_TEMPLATE/rc.md:1]: A check for renaming a cluster is already there. Workspaces are not covered in this PR.

**Pull request merged**